torch/onnx/symbolic_opset11.py

```diff
@@ -180,6 +180,60 @@ def round(g, self):
     return g.op("Round", self)
 
 
+# Generate paddings in ONNX order based on pad in pytorch.
+# Arguments:
+#     dim: the dimension of the tensor.
+#     pad: the paddings in pytorch.
+#          The order is dim_n_begin, dim_n_end, dim_n-1_begin, dim_n-1_end, ...
+def _prepare_onnx_paddings(g, dim, pad):
```

Comment:
The logic is fine here. This supports paddings of arbitrary or unknown shape. A few suggestions.

1. Please add more comments with examples illustrating what each step is doing.
2. `Pad` can be replaced with `Concat`. This should make the code cleaner as it would get rid of the Casting.
3. Fuse the last step `Reshape` to [1, -1], and `Squeeze` [0] to `Reshape` to [-1].
4. Use `torch.onnx.symbolic_opset9.size` to get pad_len.

Reply:
Thanks for the suggestion. As we talked, I updated this Concat + ConstantOfShape.